examples/platforms/nrf52840/diag.c

```diff
@@ -28,21 +28,194 @@
+
+    if (argc == 0)
+    {
+        snprintf(aOutput, aOutputMaxLen, "ID: %" PRId16 "\r\n", sID);
+    }
+    else
+    {
+        long value;
+
+        SuccessOrExit(error = parseLong(argv[0], &value));
+        VerifyOrExit(value >= 0, error = kThreadError_InvalidArgs);
+        sID = (int16_t)(value);
+        snprintf(aOutput, aOutputMaxLen, "set ID to %" PRId16 "\r\nstatus 0x%02x\r\n", sID, error);
+    }
+
+exit:
+    appendErrorResult(error, aOutput, aOutputMaxLen);
+}
+
+static void processTransmit(otInstance *aInstance, int argc, char *argv[], char *aOutput,
+                            size_t aOutputMaxLen)
+{
+    ThreadError error = kThreadError_None;
+
+    VerifyOrExit(otPlatDiagModeGet(), error = kThreadError_InvalidState);
+
+    if (argc == 0)
+    {
+        snprintf(aOutput, aOutputMaxLen, "transmit will send %" PRId32 " diagnostic messages with %" PRIu32
+                 " ms interval\r\nstatus 0x%02x\r\n",
+                 sTxRequestedCount, sTxPeriod, error);
+    }
+    else if (strcmp(argv[0], "stop") == 0)
+    {
+        otPlatAlarmStop(aInstance);
+        snprintf(aOutput, aOutputMaxLen, "diagnostic message transmission is stopped\r\nstatus 0x%02x\r\n", error);
+        sTransmitActive = false;
+    }
+    else if (strcmp(argv[0], "start") == 0)
+    {
+        otPlatAlarmStop(aInstance);
+        sTransmitActive = true;
+        sTxCount = sTxRequestedCount;
+        uint32_t now = otPlatAlarmGetNow();
+        otPlatAlarmStartAt(aInstance, now, sTxPeriod);
+        snprintf(aOutput, aOutputMaxLen, "sending %" PRId32 " diagnostic messages with %" PRIu32
+                 " ms interval\r\nstatus 0x%02x\r\n",
+                 sTxRequestedCount, sTxPeriod, error);
+    }
+    else if (strcmp(argv[0], "interval") == 0)
+    {
+        long value;
 
-void otPlatDiagProcess(int argc, char *argv[], char *aOutput, size_t aOutputMaxLen)
+        VerifyOrExit(argc == 2, error = kThreadError_InvalidArgs);
+
+        SuccessOrExit(error = parseLong(argv[1], &value));
+        VerifyOrExit(value > 0, error = kThreadError_InvalidArgs);
+        sTxPeriod = (uint32_t)(value);
+        snprintf(aOutput, aOutputMaxLen, "set diagnostic messages interval to %" PRIu32 " ms\r\nstatus 0x%02x\r\n", sTxPeriod,
+                 error);
+    }
+    else if (strcmp(argv[0], "count") == 0)
+    {
+        long value;
+
+        VerifyOrExit(argc == 2, error = kThreadError_InvalidArgs);
+
+        SuccessOrExit(error = parseLong(argv[1], &value));
+        VerifyOrExit((value > 0) || (value == -1), error = kThreadError_InvalidArgs);
+        sTxRequestedCount = (uint32_t)(value);
+        snprintf(aOutput, aOutputMaxLen, "set diagnostic messages count to %" PRId32 "\r\nstatus 0x%02x\r\n", sTxRequestedCount,
+                 error);
+    }
+
+exit:
+    appendErrorResult(error, aOutput, aOutputMaxLen);
+}
+
+const struct PlatformDiagCommand sCommands[] =
+{
+    {"listen", &processListen },
+    {"transmit", &processTransmit },
```

Comment:
Yes they have similar functionality but neither `diag send` nor 'diag repeat' sends data that we needed so I've implemented this platform specific version.